Source/WebCore/page/scrolling/ScrollAnchoringController.cpp

```diff
@@ -112,17 +112,17 @@ void ScrollAnchoringController::notifyChildHadSuppressingStyleChange()
 
 bool ScrollAnchoringController::isInScrollAnchoringAncestorChain(const RenderObject& object)
 {
-    RefPtr iterElement = m_anchorElement.get();
+    if (!m_anchorElement)
+        return false;
 
+    auto* iterElement = m_anchorElement->renderer();
     while (iterElement) {
-        if (auto* renderer = iterElement->renderer()) {
-            LOG_WITH_STREAM(ScrollAnchoring, stream << "ScrollAnchoringController::isInScrollAnchoringAncestorChain() checking for : " <<object << " current Element: " << *iterElement);
-            if (&object == renderer)
-                return true;
-        }
-        if (iterElement && elementIsScrollableArea(*iterElement, m_owningScrollableArea))
+        LOG_WITH_STREAM(ScrollAnchoring, stream << "ScrollAnchoringController::isInScrollAnchoringAncestorChain() checking for : " <<object << " current Element: " << *iterElement);
+        if (&object == iterElement)
+            return true;
+        if (iterElement && elementIsScrollableArea(*iterElement->element(), m_owningScrollableArea))
             break;
-        iterElement = iterElement->parentElement();
```

Comment:
Can we just switch to `parentInComposedTree()` instead?

Reply:
parentElementInComposedTree works here too

Reply:
Changed to `parentElementInComposedTree`.